### PR TITLE
Fix Erlang code generation for floating point numbers

### LINKED ISSUE #3073: Improper Erlang Codegen of Scientific Notation Floats without Trailing Zeros
```sh
> gleam --version
> gleam 1.1.0
```

Consider the following:
```gleam
pub fn main() {
  1.0
  1.
  1.0e6
  1.e6
}
```
The last expression (`1.e6`) causes an error in escript:

```sh
~/gleam/float_test > gleam run --target erlang
  Compiling float_test

~/gleam/float_test/build/dev/erlang/float_test/_gleam_artefacts/float_test.erl:11:6: syntax error before: '.'
%   11|     1.e6.
%     |      ^

~/gleam/float_test/build/dev/erlang/float_test/_gleam_artefacts/float_test.erl:4:2: function main/0 undefined
%    4| -export([main/0]).
%     |  ^

~/gleam/float_test/build/dev/erlang/float_test/_gleam_artefacts/float_test.erl:6:2: spec for undefined function main/0
%    6| -spec main() -> float().
%     |  ^

error: Shell command failure

There was a problem when running the shell command `escript`.
```

I'm not familiar with erlang, but I assume that floats without trailing zeroes is a gleam thing, as both `1.` and `1.e6` fail to parse in `erl`.

This does not fail when `--target js`.

### PR DESCRIPTION
**Pull request by giacomocavalieri**
This PR fixes #3073

### CODE REVIEW

**Comment by lpil** [comment]
Snaps need to be updated!

**Comment by giacomocavalieri** [comment]
Oops, fixed!

File: compiler-core/src/javascript/pattern.rs [hunk 1]
```diff
@@ -20,6 +20,7 @@ enum Index<'a> {
     StringPrefixSlice(usize),
 }
 
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (lpil) [reply compiler-core/src/javascript/pattern.rs:23]: Looks like it's not used! Let's remove it

**Review by lpil — APPROVED** [review]
Thank you!!

**Pull request merged**